metricbeat/module/envoyproxy/server/data.go

```diff
@@ -0,0 +1,187 @@
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License
+
+package server
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstrstr"
+)
+
+var (
+	schema = s.Schema{
+		"cluster_manager": s.Object{
+			"active_clusters":  c.Int("active_clusters"),
+			"cluster_added":    c.Int("cluster_added"),
+			"cluster_modified": c.Int("cluster_modified"),
+			"cluster_removed":  c.Int("cluster_removed"),
+			"warming_clusters": c.Int("warming_clusters"),
+		},
+		"filesystem": s.Object{
+			"flushed_by_timer":     c.Int("flushed_by_timer"),
+			"reopen_failed":        c.Int("reopen_failed"),
+			"write_buffered":       c.Int("write_buffered"),
+			"write_completed":      c.Int("write_completed"),
+			"write_total_buffered": c.Int("write_total_buffered"),
+		},
+		"runtime": s.Object{
+			"admin_overrides_active":  c.Int("admin_overrides_active"),
+			"load_error":              c.Int("load_error"),
+			"load_success":            c.Int("load_success"),
+			"num_keys":                c.Int("num_keys"),
+			"override_dir_exists":     c.Int("override_dir_exists"),
+			"override_dir_not_exists": c.Int("override_dir_not_exists"),
+		},
+		"listener_manager": s.Object{
+			"listener_added":           c.Int("listener_added"),
+			"listener_create_failure":  c.Int("listener_create_failure"),
+			"listener_create_success":  c.Int("listener_create_success"),
+			"listener_modified":        c.Int("listener_modified"),
+			"listener_removed":         c.Int("listener_removed"),
+			"total_listeners_active":   c.Int("total_listeners_active"),
+			"total_listeners_draining": c.Int("total_listeners_draining"),
+			"total_listeners_warming":  c.Int("total_listeners_warming"),
+		},
+		"stats": s.Object{
+			"overflow": c.Int("overflow"),
+		},
+		"server": s.Object{
+			"days_until_first_cert_expiring": c.Int("days_until_first_cert_expiring"),
+			"live":               c.Int("live"),
+			"memory_allocated":   c.Int("memory_allocated"),
+			"memory_heap_size":   c.Int("memory_heap_size"),
+			"parent_connections": c.Int("parent_connections"),
+			"total_connections":  c.Int("total_connections"),
+			"uptime":             c.Int("uptime"),
+			"version":            c.Int("version"),
+			"watchdog_mega_miss": c.Int("watchdog_mega_miss"),
+			"watchdog_miss":      c.Int("watchdog_miss"),
+		},
+		"listener": s.Object{
```

Comment:
Is the list of `listeners` and `http` objects dynamic? If it is maybe `Schema` (as is now) is not the best option to parse these fields.

Reply:
What do you recommend to parse these fields? Do you have an example for this?

Reply:
Thank you for the reply! I removed dynamic fields and updated docs for the non-dynamic fields.  In the meantime, I will look to other modules for dynamic fields.